uvls/src/main.rs

```diff
@@ -394,22 +396,36 @@ impl LanguageServer for Backend {
             "uvls/generate_diagram" => {
                 let root_fileid = FileID::from_uri(&Url::parse(uri.as_str()).unwrap());
                 let root_graph = self.pipeline.root().borrow_and_update().clone();
-                if !root_graph.contains_id(root_fileid){{}}
-                
+                if !root_graph.contains_id(root_fileid) {
+                    {}
+                }
+
                 let document = root_graph.files.get(&root_fileid).unwrap();
-                let g = ast::graph::Graph::new(document.source.clone(), document.tree.clone(), uri.clone());
+                let g = ast::graph::Graph::new(
+                    document.source.clone(),
+                    document.tree.clone(),
+                    uri.clone(),
+                );
 
                 // write graph script (dot) to file:
                 let diagram_file_extension = "dot";
                 let re = regex::Regex::new(r"(.*\.)(.*)").unwrap();
-                let path = re.replace(uri.path(), |caps: &regex::Captures| {format!("{}{}", &caps[1], diagram_file_extension)});
-                let file = std::fs::File::create(path.as_ref())
-                    .or(std::fs::File::create(percent_decode_str(&path.replacen("/", "", 1)).decode_utf8().unwrap().into_owned())); // windows specific
+                let path = re.replace(uri.path(), |caps: &regex::Captures| {
+                    format!("{}{}", &caps[1], diagram_file_extension)
+                });
+                let file = std::fs::File::create(path.as_ref()).or(std::fs::File::create(
```

Comment:
same here.  
this part is probably not interesting for editors in the future and therefore can be written in one line of code imo.